crates/astria-core/src/protocol/genesis/v1.rs

```diff
@@ -46,12 +47,7 @@ use crate::{
     Protobuf,
 };
 
-#[derive(Clone, Debug)]
-#[cfg_attr(
-    feature = "serde",
-    derive(serde::Deserialize, serde::Serialize),
-    serde(try_from = "raw::ConnectGenesis", into = "raw::ConnectGenesis")
-)]
+#[derive(Clone, Debug, BorshSerialize)]
```

Comment:
I Am surprised to see borsh as part of astria-core in general. So far my view of astria-core was for it to be a) a collection of domain types, and b) the conversion from/to on-wire types (protobuf). Borsh on the other hand was intended to be the on-disk format and an implementation detail of sequencer.

Reply:
Yes - that's up for debate.

It's used to encode the data comprising the upgrade changes in order to get a hash of each change.  This needs to be deterministic so protobuf encoding isn't suitable.

Perhaps best to discuss this elsewhere and I can update here with the conclusion.